### ISSUE TITLE
Problem with JSC on arm64 environment

### ISSUE THREAD

**Issue by cujo30227** [phoboslab/Ejecta#457]
Hi,
When I try to use the JSC framework in the Ejecta project in a 64bit compiler, I get the error messages shown below. Apparently the framework does not contain 64 bit slices?
ld: warning: ignoring file XXXXXX/JavaScriptCore.framework/JavaScriptCore, missing required architecture x86_64 in file XXXXX/JavaScriptCore.framework/JavaScriptCore (4 slices)
Undefined symbols for architecture x86_64:
  "_JSContextGetGlobalObject", referenced from:

When I compile a JSC framework myself, and add this to my project, the number of errors reduces, but I still get the following 4 errors:

Undefined symbols for architecture x86_64:
  "std::__1::__call_once(unsigned long volatile&, void_, void (_)(void_))", referenced from:
      JSC::DFG::globalWorklist() in JavaScriptCore(DFGWorklist.o)
      JSC::initializeThreading() in JavaScriptCore(InitializeThreading.o)
      WTF::exchangeIsCompilationThread(bool) in JavaScriptCore(CompilationThread.o)
  "void std::__1::__sort<std::__1::__less<unsigned int, unsigned int>&, unsigned int_>(unsigned int_, unsigned int_, std::__1::__less<unsigned int, unsigned int>&)", referenced from:
      JSC::DFG::Graph::dumpBlockHeader(WTF::PrintStream&, char const_, JSC::DFG::BasicBlock_, JSC::DFG::Graph::PhiNodeDumpMode, JSC::DumpContext_) in JavaScriptCore(DFGGraph.o)
      JSC::JSObject::getOwnPropertyNames(JSC::JSObject_, JSC::ExecState_, JSC::PropertyNameArray&, JSC::EnumerationMode) in JavaScriptCore(JSObject.o)
      JSC::computePreciseJumpTargets(JSC::CodeBlock_, WTF::Vector<unsigned int, 32ul, WTF::CrashOnOverflow>&) in JavaScriptCore(PreciseJumpTargets.o)
  "void std::__1::__sort<std::__1::__less<unsigned long, unsigned long>&, unsigned long_>(unsigned long_, unsigned long*, std::__1::__less<unsigned long, unsigned long>&)", referenced from:
      JSC::DFG::NaturalLoops::compute(JSC::DFG::Graph&) in JavaScriptCore(DFGNaturalLoops.o)
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)

Can anybody help me? Also, does someone have a static library available (libJavaScriptCore.a) with a arm64 slice in it?

Thanks and happy holidays!

**Comment by phoboslab** [comment]
I believe this has something to do with the project settings. Maybe the C/C++ compilation mode and the included standard libraries!?

Try if re-downloading an compiling without changing the project settings works. It does for me :)

**Comment by cujo30227** [comment]
Hi, it works now. It turned out that I needed to add the libcucore.dylib and libc++.dylib libraries to my project. BTW> after releasing the 64 bits version of my app with this JavaScriptCore framework, I suddenly received a lot more SIGSEGV crashes related to JSEvaluateScript. Below the last 4 lines of the stacktrace. I do call JSEvaluateScript in my code like this: JSEvaluateScript(ctx, scriptJS, NULL, NULL, 1, &exception);  I tried everything, but I cannot get this function to crash in order to reproduce the crash. Anybody has any ideas what could make JSEvaluateScript crash??
Thanks for the excellent work!!!

0x00339b36 WTFCrash (Assertions.cpp:341)
0x00339af5 WTFPrintBacktrace (Assertions.cpp:313)
0x0013311f JSC::evaluate(JSC::ExecState*, JSC::SourceCode const&, JSC::JSValue, JSC::JSValue*) + 596
0x00229251 JSEvaluateScript + 266
